department-of-veterans-affairs/vets-website
# [MHV-51044] Success message(s) banners need focus

## Summary

- Adjusted the screenreader focus to land on the alert if one is visible within the FolderThreadView container.
- This is to prevent an unexpected shift in focus on the h1
Scenarios of occurrence: 
- When a message or reply message is SENT, MOVED, or DELETED from:
- [x] INBOX
- [x] DRAFTS
- [x] SENT
- [x] TRASH
- [x] ANY CUSTOM FOLDER

- When a custom folder is CREATED, RENAMED or REMOVED

## Related issue(s)

### [MHV-51044](https://jira.devops.va.gov/browse/MHV-51044) - Success message(s) banners need focus
>Inbox, drafts, sent, trash, and my folders interactions alerts
>Whenever a person sends a message, deletes a draft, or moves a folder (and so on), it takes the person to a page that displays a successful banner and immediately shifts the focus to the H1.
>
>Problem
>This unexpected shift in focus to the H1 element results in confusion and a higher likelihood of people missing the notification of successful actions, particularly when using screen readers.
>
>Solution
>To fix this finding, it is recommended to not have the focus change to the H1. Instead, keep the focus on the cancel button within the success banner and apply aria-live with a polite setting to announce the message.
>
>Note: the images show the current state of the alert(s) cancel button not receiving >focus, instead it shifts to the heading level 1.
>
>User Story As a SM screen reader user, I want the screen reader to announce that my message has been sent.
>
>GIVEN: Screen reader users types a SM message
>WHEN: message is sent successfully
>THEN: Screen reader announces that the message was sent successfully
>
>Feature Flag Y/N ? N
>Manual Testing Y/N ? Y
>Automated Testing Y/N ? Y
>Accessibility Testing Y/N ? Y
>
>Acceptance Criteria:
>AC1 Screen reader announces that a message has been sent successfully
>AC2 Focus should highlight the success message
>AC3 All flows that send a message should be covered, reply, send, and delete draft, >saving a draft, moving to a different folder
>AC4
>
## Testing done


## Screenshots
**N/A - ALLY UPDATES

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        |       |
| Desktop |        |       |

## What areas of the site does it impact?

*Va.gov - Secure Messaging*

## Acceptance criteria

### Quality Assurance & Testing


### Error Handling


### Authentication


### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:


## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_
